### PR TITLE
clippy: fix warnings in components/net

### PR DESCRIPTION
**Pull request by eerii**
Split from #31514, fixes some clippy warnings in `components/net`.

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes are part of #31500
- [x] These changes do not require tests because they do not modify functionality, they only fix lint errors.

### CODE REVIEW

File: components/net/http_loader.rs [hunk 1]
```diff
@@ -101,13 +99,19 @@ pub struct HttpState {
     pub override_manager: CertificateErrorOverrideManager,
 }
 
+impl Default for HttpState {
+    fn default() -> Self {
+        Self::new()
```
Comment (mrobinson) [inline components/net/http_loader.rs:104]: Can the `new()` method be removed now?

File: components/net/http_cache.rs [hunk 2]
```diff
@@ -586,6 +582,12 @@ fn handle_range_request(
     None
 }
 
+impl Default for HttpCache {
+    fn default() -> Self {
+        Self::new()
```
Comment (mrobinson) [inline components/net/http_cache.rs:587]: Can the `new()` method be removed now?

File: components/net/http_loader.rs [hunk 3]
```diff
@@ -1917,11 +1921,11 @@ async fn http_network_fetch(
...
-    if let Some(encoding) = response.headers.typed_get::<ContentEncoding>() {
-        if encoding.contains("gzip") {
-        } else if encoding.contains("compress") {
-        }
-    };
+    // if let Some(encoding) = response.headers.typed_get::<ContentEncoding>() {
+    //     if encoding.contains("gzip") {
+    //     } else if encoding.contains("compress") {
+    //     }
+    // };
```
Comment (mrobinson) [inline components/net/http_loader.rs:1928]: It looks like the bug is resolved so I would just remove all of Step 12 and this code as well.

**Review by mrobinson — CHANGES_REQUESTED** [review]
Nice! Just a couple comments below:

**Review by mrobinson — APPROVED** [review]

File: components/net/image_cache.rs [hunk 4]
```diff
@@ -224,7 +218,7 @@ impl ImageBytes {
                 ImageBytes::InProgress(ref mut bytes) => bytes,
                 ImageBytes::Complete(_) => panic!("attempted modification of complete image bytes"),
             };
-            mem::replace(own_bytes, vec![])
+            std::mem::take(own_bytes)
```
Comment (Loirooriol) [inline components/net/image_cache.rs:221]: Either `mem::take` or stop importing `std::mem`, but this is causing a compile warning (#31607).

Comment (eerii) [reply components/net/image_cache.rs:221]: Sorry about that, I'll fix it right away.

Comment (eerii) [reply components/net/image_cache.rs:221]: Should be fixed with #31608

**Pull request merged**